src/pivot-table/components/cells/utils/snapshot-scroll-calculations.ts

```diff
@@ -0,0 +1,84 @@
+import type { PivotLayout } from "../../../../types/QIX";
+import type { ViewService } from "../../../../types/types";
+
+const getCellOfLeftColumn = (
+  dataYindex: number,
+  cells?: NodeListOf<Element>,
+): { el: Element; index: number } | undefined => {
+  if (!cells) return undefined;
+  for (let i = 0; i < cells.length; i++) {
+    const dataY = cells[i].getAttribute("data-y");
+    if (dataY && parseInt(dataY, 10) === dataYindex)
+      return {
+        el: cells[i],
+        index: i,
+      };
+  }
+  return undefined;
+};
+
+const findIndex = (
+  position: "Start" | "End",
+  cell: { el: Element; index: number } | undefined,
+  bodyRect: DOMRect | undefined,
+): number | undefined => {
+  const cellRect = cell?.el?.getBoundingClientRect();
+  if (!cellRect) return -1;
+  if (position === "Start" && bodyRect && cellRect.bottom > bodyRect.top) return cell?.index;
+  if (position === "End" && bodyRect && bodyRect.bottom > cellRect.top) return cell?.index;
+  return -1;
+};
+
+export const getPartialTopScrollHeight = (
+  cell: { el: Element; index: number } | undefined,
+  bodyRect: DOMRect | undefined,
+) => {
+  if (!cell) return 0;
+  const cellRect = cell?.el?.getBoundingClientRect();
+  return bodyRect && Math.max(bodyRect.top - cellRect.top, 0);
+};
+
+export const findVisibleRow = (viewService: ViewService, element: HTMLElement) => {
+  const leftColumnBody = element.querySelector(
```

Comment:
I think you always want to use the "Data grid", since that will always be there even if you don't have any left dimensions